File: python/structural/decorator/decorators/message_decorator.py
from message.message import IMessage

class MessageDecorator(IMessage):
    def __init__(self, message: IMessage):
        self.message = message

    def MessageDecorator(self, message: IMessage):
        self.message = message

    def set_content(self, content: str):
        self.message.set_content(content)
    
    def get_content(self) -> str:
        return self.message.get_content()

File: typescript/src/structural/decorator/decorators/MessageDecorator.ts
import { IMessage } from "../interfaces/IMessage";

export class MessageDecorator implements IMessage{
    protected message: IMessage

    constructor(message: IMessage){
        this.message = message
    }

    processMessage(): IMessage {
        return this.message.processMessage()
    }

    getContent(): string {
        return this.message.getContent()
    }

    setContent(content: string): void {
        this.message.setContent(content)
    }


}